platform/linux-generic/odp_packet.c

```diff
@@ -354,6 +354,8 @@ static inline void link_segments(odp_packet_hdr_t *pkt_hdr[], int num)
 			hdr->buf_hdr.seg[i].hdr  = buf_hdr;
 			hdr->buf_hdr.seg[i].data = buf_hdr->base_data;
 			hdr->buf_hdr.seg[i].len  = BASE_LEN;
+			ODP_ASSERT((odp_atomic_store_u32(&hdr->buf_hdr.ref_cnt,
```

Comment:
Bill, odp_atomic_store_u32() is void. How assert() works here?

Reply:
It's a compound statement. One of the clever bits of Petri's rework is he doesn't set the initial `ref_cnt` to 1 but rather leaves it as 0 and when the first reference happens bumps it directly to 2. For debugging purposes we want to ensure that the `ref_cnt >= 1` on any free so we need the initial `ref_cnt` to be set to 1, but only for debug builds. The compound statement takes advantage of the fact that in C the "value" of a compound statement is the value of the last component, in this case 1. So the `ODP_ASSERT()` always succeeds and the effect is that the `ref_cnt` is set to 1 only if we're running with `--enable-debug`.

Reply:
This is bad, bad, bad idea. Contents of ODP_ASSERT can disappear. So `odp_atomic_store_u32` won't be called.

Reply:
@muvarov No, it unrolls to ` ... && (void_func(), 1) ...`. Comma operator.
@Bill-Fischofer-Linaro such asserts need to be reworked.

Reply:
@lumag yep, but what is (void_func(), 1) ? Why gcc compiles that?